test/js/applesoft/compiler.spec.ts

```diff
@@ -0,0 +1,431 @@
+import ApplesoftCompiler from 'js/applesoft/compiler';
+import RAM from 'js/ram';
+import { Memory } from 'js/types';
+
+// Zero page locations used by Applesoft. The names come from
+// the commented decompilation produced by the Merlin Pro
+// assembler, revision 4/27/84. There is evidence from
+// https://www.pagetable.com/?p=774 that the original Microsoft
+// BASIC source code used these names as well.
+const TXTTAB = 0x67; // start of program, word
+const VARTAB = 0x69; // start of variables, word
+const ARYTAB = 0x6B; // start of arrays, word
+const STREND = 0x6D; // end of strings, word
+const PRGEND = 0xAF; // end of program, word
```

Comment:
Most of these get reused elsewhere, maybe export from here or a constants file in this directory?